local_planner/include/local_planner/trajectory_simulator.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <eigen3/Eigen/Core>
+
+namespace avoidance {
+
```

Comment:
Just curious, what's the rationale behind the architecture here? Why individual functions as opposed to wrapping this in a class? Maybe it's just a brief draft and not thought through yet, maybe there's a reason I'm not aware of. I guess eventually we would want to include it in the planner through composition, as a 'Simulator' object or similar?

Reply:
Good idea, I've arranged it to share the setup of the simulator so it can be initialized at each node.